presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergFileWriterFactory.java

```diff
@@ -133,28 +126,43 @@ private IcebergFileWriter createParquetWriter(
             JobConf jobConf,
             ConnectorSession session)
     {
-        Properties properties = new Properties();
-        properties.setProperty(IOConstants.COLUMNS, columns.stream()
-                .map(IcebergColumnHandle::getName)
-                .collect(joining(",")));
-        properties.setProperty(IOConstants.COLUMNS_TYPES, columns.stream()
-                .map(column -> toHiveType(column.getType()).getHiveTypeName().toString())
-                .collect(joining(":")));
-
-        setParquetSchema(jobConf, convert(icebergSchema, "table"));
-        jobConf.set(ParquetOutputFormat.COMPRESSION, getCompressionCodec(session).getParquetCompressionCodec().name());
-
-        return new IcebergRecordFileWriter(
-                outputPath,
-                columns.stream()
-                        .map(IcebergColumnHandle::getName)
-                        .collect(toImmutableList()),
-                fromHiveStorageFormat(HiveStorageFormat.PARQUET),
-                properties,
-                HiveStorageFormat.PARQUET.getEstimatedWriterSystemMemoryUsage(),
-                jobConf,
-                typeManager,
-                session);
+        List<String> fileColumnNames = icebergSchema.columns().stream()
+                .map(Types.NestedField::name)
+                .collect(toImmutableList());
+        List<Type> fileColumnTypes = icebergSchema.columns().stream()
+                .map(column -> toPrestoType(column.type(), typeManager))
+                .collect(toImmutableList());
+
+        int[] fileInputColumnIndexes = IntStream.range(0, fileColumnNames.size()).toArray();
```

Comment:
Let's inline this, to match the ORC code

Reply:
Done.